subsys/net/ip/tcp2.c

```diff
@@ -834,6 +940,45 @@ static struct tcp *tcp_conn_new(struct net_pkt *pkt)
 	return conn;
 }
 
+static void update_acked_data(struct tcp *conn, struct tcphdr *th)
+{
+	if (!th) {
+		return;
+	}
+
+	if (!net_tcp_seq_greater(th_ack(th), conn->seq)) {
+		/* If there is any pending data, remove them
+		 * from re-send as peer just acked it
+		 */
+		u32_t acked_len = th_ack(th) - conn->last_seq;
+		int ret;
+
+		if (conn->pending_data && acked_len) {
```

Comment:
maybe test the contrary and return if it matches. Then the code block below will be indented less (there is 4 nested if now)

Reply:
Sure, that makes sense.